code/modules/mob/living/simple_animal/simple_animal.dm

```diff
@@ -109,25 +117,36 @@
 /mob/living/simple_animal/proc/setup_languages()
 	add_language(/decl/language/animal)
 
-/mob/living/simple_animal/proc/check_mob_icon_states()
-	mob_icon_state_flags = 0
-	if(check_state_in_icon("world", icon))
-		mob_icon_state_flags |= MOB_ICON_HAS_LIVING_STATE
-	if(check_state_in_icon("world-dead", icon))
-		mob_icon_state_flags |= MOB_ICON_HAS_DEAD_STATE
-	if(check_state_in_icon("world-sleeping", icon))
-		mob_icon_state_flags |= MOB_ICON_HAS_SLEEP_STATE
-	if(check_state_in_icon("world-resting", icon))
-		mob_icon_state_flags |= MOB_ICON_HAS_REST_STATE
-	if(check_state_in_icon("world-gib", icon))
-		mob_icon_state_flags |= MOB_ICON_HAS_GIB_STATE
+var/global/list/simplemob_icon_bitflag_cache = list()
+/mob/living/simple_animal/proc/check_mob_icon_states(var/sa_initializing = FALSE)
+	if(sa_initializing) // Let people force-rebuild the mob cache with proccall if needed.
+		mob_icon_state_flags = global.simplemob_icon_bitflag_cache[type]
+	else
+		mob_icon_state_flags = null
+	if(isnull(mob_icon_state_flags))
+		mob_icon_state_flags = 0
+		if(check_state_in_icon("world", icon))
+			mob_icon_state_flags |= MOB_ICON_HAS_LIVING_STATE
+		if(check_state_in_icon("world-dead", icon))
+			mob_icon_state_flags |= MOB_ICON_HAS_DEAD_STATE
+		if(check_state_in_icon("world-sleeping", icon))
+			mob_icon_state_flags |= MOB_ICON_HAS_SLEEP_STATE
+		if(check_state_in_icon("world-resting", icon))
+			mob_icon_state_flags |= MOB_ICON_HAS_REST_STATE
+		if(check_state_in_icon("world-gib", icon))
+			mob_icon_state_flags |= MOB_ICON_HAS_GIB_STATE
+		if(check_state_in_icon("world-dying", icon))
+			mob_icon_state_flags |= MOB_ICON_HAS_DYING_STATE
+		global.simplemob_icon_bitflag_cache[type] = mob_icon_state_flags
 
 /mob/living/simple_animal/on_update_icon()
 
 	..()
 
 	icon_state = ICON_STATE_WORLD
-	if(stat == DEAD && (mob_icon_state_flags & MOB_ICON_HAS_DEAD_STATE))
+	if(is_aquatic && stat != DEAD && HAS_STATUS(src, STAT_PARA) && (mob_icon_state_flags & MOB_ICON_HAS_DYING_STATE))
+		icon_state += "-dying"
```

Comment:
Why gate this behind aquatic, as opposed to just letting the `HAS_STATUS` figure it out?